packages/create-cloudflare/src/templates.ts

```diff
@@ -65,16 +65,21 @@ export type TemplateConfig = {
 	 * to handle config version skew between different versions of c3
 	 */
 	configVersion: number;
-	/** The id by which template is referred to internally and keyed in lookup maps*/
+	/** The id by which template is referred to internally and keyed in lookup maps */
 	id: string;
-	/** A string that controls how the template is presented to the user in the selection menu*/
+	/** A string that controls how the template is presented to the user in the selection menu */
 	displayName: string;
-	/** A string that explains what is inside the template, including any resources and how those will be used*/
+	/** A string that explains what is inside the template, including any resources and how those will be used */
 	description?: string;
 	/** The deployment platform for this template */
 	platform: "workers" | "pages";
-	/** The name of the framework cli tool that is used to generate this project or undefined if none. */
+	/** The name of the framework cli tool that is used to generate this project or undefined if none */
 	frameworkCli?: string;
+	/**
+	 * The version of the framework cli tool to use.
+	 * If omitted the cli version is taken from src/frameworks/package.json, which is the default/standard behavior.
+	 */
+	frameworkCliVersion?: string;
```

Comment:
Another option would be to allow `frameworkCli` to specify the version (i.e. `create-next-app@14.2.5`)

Reply:
I think this approach is fine. You could even change it to something to make it clear this is exceptional:

```suggestion
	pinFrameworkCliVersion?: string;
```
We use dependabot to keep the CLI versions up to date, which is why they are stored in a package.json. Moving the versions here would mess that up. But in this case we don't want dependabot to do anything with this version since we are explictly pinning it.

Reply:
My proposal is to use either `cli` to use the version from the package.json or `cli@version` to pin the version as an alternative to using the `pinFrameworkCliVersion ` property

Reply:
@petebacondarwin I've renamed the field as you suggested 🙂 

I've also updated the comment to make the thing a bit clearer

https://github.com/cloudflare/workers-sdk/pull/7241/commits/c36670d831a9ea7ade96b91af8396474ea821b98